test/system/users_test.rb

```diff
@@ -0,0 +1,12 @@
+require "application_system_test_case"
+
+class UsersTest < ApplicationSystemTestCase
+  test "preserve return_to while switching from login to signup" do
+    visit '/login?return_to=/subscribe/multiple/tag/tag1,tag2'
+    click_on "sign up"
+    assert_redirected_to '/signup?return_to=/subscribe/multiple/tag/tag1,tag2'
```

Comment:
@siaw23 @gauravano @jywarren can you please help me why this line shows error?

Reply:
line 7 i mean

Reply:
Line 7 is an assertion. Which error do you mean @SidharthBansal? Can you please paste it here?

Reply:
Right. So you don't do assertion here like this. When you `click_on "sign up"` you have to assert the presence of some text on that page after clicking. So try manually clicking on "sign up" then assert the presence of some text there like so `assert_text 'Some Text'` where "Some Text" is some unique text that you can only see after clicking "sign up". Let's try that if that works.

Reply:
Test will fail as we don't have access to Recaptcha feature in the testing environment.
So, we will not be able to submit the form by clicking on the  "Sign Up" button.

Reply:
Can we try unit test for this? We just want three steps.
1. User clicks on '/login?return_to=X'
2. User clicks on "Signup"
3. User is now at '/signup?return_to=X'
But I don't know how to do statement 2 in unit tests.